clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -234,10 +234,17 @@ public RuntimeException lastError() {
     }
 
     public synchronized boolean ensurePartitionAdded(TopicPartition tp) {
```

Comment:
Added special tests for this is overkill since it depends so much on the transaction manager state. I have instead added assertions about the expected return values of this function in various existing tests, which achieves the same purpose.